### ISSUE TITLE
Get language of translation for a certain key

### ISSUE THREAD

**Issue by c7hm4r** [i18next/i18next#1763]
## 🚀 Feature Proposal

Expose a public API which does the same like the `t` function but also returns the language of the returned translation text.

## Motivation

To recap: For a key, `i18next` dynamically chooses the translation language depending on the configuration of the language resources priority (i.e. `i18n.languages`) and the existence of the key in the specified resources. Thus, when one does `translation = t("some_key")` (usually) a string is returned whose language depends on different factors.

Depending on the language of the returned `translation` string, the code should behave differently. E.g. it should set for each paragraph the right `lang` HTML attribute. However, `t("some_key")` can currently not be configured with options to also return the language code. There are two different approaches how to get the language of a translation by a key:

[Credits – a StackOverflow question](https://stackoverflow.com/questions/72120443/how-to-obtain-the-language-of-the-string-returned-by-i18next-t/)

### Current approach 1: Reimplement the wheel

```typescript
const language = i18n.languages.find((language) => i18n.exists("some_key", {lng: language}));
```

This approach is bad because it reimplements something which happens already inside the `t` function call. So it consumes unnecessary computation time. Furthermore, it could be in the future that the internal implementation of the `t` function changes such that the behavior is different. This approach is thus neither robust nor maintainable. It is incompatible with the `i18n.getFixedT` API.

### Current approach 2: Exploit internal API

```typescript
const { res,  usedLng } = (i18n as any).translator.resolve("some_key") as { res: unknown, usedLng: string };
```

Not maintainable because it uses an internal API. No TypeScript support (`any` and `as` turn the TypeScript compiler off and on). Does not provide the text in the same way `t` does (it seems to not respect `returnObjects` and does not automatically switch to the key as last fallback).

### Current approach 3: Bloat all resources

1. Configure `returnObjects: true`.
2. Replace each translation entry `"the text"` in all resources (`"en"`) by `{ language: "en", text: "the text" }` programmatically.
3. Use 

    ```typescript
    const tResult = t("some_key") as string | { language: string, text: string };
    const { language, text } = typeof tResult === "string" ? { language: "unknown", text: tResult } : tResult;
    ```
Bloat; several tedious corner cases; no TypeScript support.

## Example

```typescript
const { text, usedLanguage } = i18n.resolve("some_key");
return `<p lang="${usedLanguage}">${text}</p>`;
```

**Comment by jamuhl** [comment]
I would add a `verbose` flag to t-options like:

`i18n.t(key, { verbose: true });` and change https://github.com/i18next/i18next/blob/master/src/Translator.js#L296 to

```js
if (options.verbose) return { result: res, resolved };
return res;
```

**Comment by adrai** [comment]
Prepared a PR, but not yet sure if this will land: https://github.com/i18next/i18next/pull/1764

**Comment by jamuhl** [comment]
or what @adrai suggest...just not sure if reusing the `t` function is somewhat typescript friendly - as it already returns an object in case of returnObjects

**Comment by adrai** [comment]
The types for returnObjects is not specified in detail I think it is only object => `export type TFunctionResult = string | object | Array<string | object> | undefined | null;`
But fur sure @pedrodurek will know.

**Comment by jamuhl** [comment]
I meant typescript users might like something more specific returned as type than object for that use case

**Comment by adrai** [comment]
@c7hm4r ok for you?

**Comment by adrai** [comment]
returnDetails option included in v21.7.0
